ain/utils/V3Keystore.py

```diff
@@ -0,0 +1,294 @@
+        uuid: bytes = None,
+    ):
+        if salt is not None:
+            self.salt = salt
+        if iv is not None:
+            self.iv = iv
+        if kdf is not None:
+            self.kdf = kdf
+        if dklen is not None:
+            self.dklen = dklen
+        if n is not None:
+            self.n = n
+        if r is not None:
+            self.r = r
+        if p is not None:
+            self.r = p
+        if c is not None:
+            self.c = c
+        if cipher is not None:
+            self.cipher = cipher
+        if uuid is not None:
+            self.uuid = uuid
+
+class V3Keystore:
+    version: int
+    id: str
+    address: str
+    crypto: Crypto
+
+    def __init__(
+        self,
+        version: int,
+        id: str,
+        address: str,
+        ciphertext: str,
+        iv: str,
+        cipher: str,
+        kdf: str,
+        kdfparams: dict,
+        mac: str
+    ):
+        self.version = version
+        self.id = id
+        self.address = address
+        self.crypto = Crypto()
+        self.crypto.ciphertext = ciphertext
+        self.crypto.cipherparams = Cipherparams()
+        self.crypto.cipherparams.iv = iv
+        self.crypto.cipher = cipher
+        self.crypto.kdf = kdf
+        self.crypto.kdfparams = kdfparams
+        self.crypto.mac = mac
+
+    @classmethod
+    def fromPrivateKey(
+        cls,
+        privateKey: Union[bytes, str],
+        password: str,
+        options: V3KeystoreOptions = V3KeystoreOptions()
+    ):
+        """
+        Converts an account into a V3 Keystore and encrypts it with a password.
+        """
+
+        if type(privateKey) is str:
+            privateKeyBytes = bytes.fromhex(privateKey)
+        elif type(privateKey) is bytes:
+            privateKeyBytes = privateKey
+        else:
+            raise ValueError("Invalid public key type")
+
+        salt = getattr(options, "salt", token_bytes(32))
+        iv = getattr(options, "iv", token_bytes(16))
+        uuid = getattr(options, "uuid", token_bytes(16))
+        dklen = getattr(options, "dklen", 32)
+        kdf = getattr(options, "kdf", "scrypt")
+
+        kdfparams = {
+            "dklen": dklen,
+            "salt": salt.hex(),
+        }
+        if kdf == "scrypt":
+            n = getattr(options, "n", 262144)
+            r = getattr(options, "r", 8)
+            p = getattr(options, "p", 1)
+            kdfparams.update({
+                "n": n,
+                "r": r,
+                "p": p,
+            })
+            derivedKeyTemp = scrypt(
+                password,
+                salt, # type: ignore
+                dklen,
+                N=n,
+                r=r,
+                p=p
+            )
+            if type(derivedKeyTemp) is bytes:
+                derivedKey = derivedKeyTemp
+            else:
+                raise ValueError("scrypt derived multiple keys")
+        elif kdf == "pbkdf2":
+            c = getattr(options, "c", 1000)
```

Comment:
is this something that ain-util needs to change? from 262144 to 1000?

Reply:
I think so, but that isn't the point of 'doesn't work'.  When I use `kdf` as `pdkdf2` on js, it produces the following error.

`/Users/kri/Project/dep/p2/node_modules/@ainblockchain/ain-util/dist/index.js:568
        derivedKey = pbkdf2Sync(Buffer.from(password), Buffer.from(kdfparams.salt, 'hex'), kdfparams.c, kdfparams.dklen, 'sha256');
                     ^

TypeError: pbkdf2Sync is not a function`

Reply:
Ah, thanks for catching the issue! I think it's an importing problem. Let me report the issue on ain-util & fix it.

Reply:
Can I see the ref that changes 262144 to 1000??

Reply:
I think ethereum still use 262144. https://github.com/ethereumjs/keythereum/blob/3a802538a7f857add6058120aa155e71837a7ec7/index.js

Reply:
Fixed, 1000 was the default value of `PyCryptodome`.